package.json

```diff
@@ -4,6 +4,7 @@
   "description": "Pin mapping functionality for the Raspberry Pi in Node.js",
   "main": "lib/index.js",
   "scripts": {
+    "build": "gulp",
```

Comment:
I actually first added it as `prepublish` but then realised that maybe that isn't what you want since you commit the compiled file.

Have you considered this setup? The compiled file, in this case `lib/index.js` is put into the `.gitignore`. It's also added to the `files` attribute in the `package.json`. Lastly, `gulp` is added to the `prepublish` script, which is a special script. In addition to running before `npm publish` (ensuring that the latest version is always the one published), it also runs after local `npm installs`.

I think it's a great way to avoid committing multiple sources of truth, and to transpile automatically before publishing...